sql/parse/parse.go

```diff
@@ -241,6 +241,9 @@ func convert(ctx *sql.Context, stmt sqlparser.Statement, query string) (sql.Node
 	case *sqlparser.DropRole:
 		return plan.NewDropRole(n.IfExists, convertAccountName(n.Roles...)), nil
 	case *sqlparser.GrantPrivilege:
+		if n.PrivilegeLevel.Database == sql.InformationSchemaDatabaseName {
+			return nil, sql.ErrDatabaseAccessDeniedForUser.New(ctx.Session.Client().User, n.PrivilegeLevel.Database)
+		}
```

Comment:
These might fit better inside the `plan.Grant` code, so that it tests this with the rest of the code that executes a grant, instead of in this parser-focused code. Similar comment for the `plan.Revoke` code below.